### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by albertoperdomo**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    params: {},
+    exp: Math.round(Date.now() / 1000) + (10 * 60) // 10 minute expiration
+  };
+  var token = jwt.sign(payload, METABASE_EMBEDDING_SECRET);
+
+  var iframeUrl = new URL("/embed/dashboard/" + token, METABASE_SITE_URL);
+  iframeUrl.hash = "bordered=true&titled=true";
+  res.send(
+    `<iframe src="${iframeUrl}" frameborder="0" width="1280" height="1000" allowtransparency></iframe>`
+  );
```
Comment (oleggromov) [inline index.js:22]: I don't know the embeddings domain well enough but this seems ok. Easy to understand - which I suppose is what required.

However, I'd change `var`s inside the callback to `const`s too.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "scripts": {
+      "test": "echo \"Error: no test specified\" && exit 1"
+    },
+    "author": "Metabase",
+    "license": "MIT",
+    "dependencies": {
+      "express": "^4.18.2",
+      "jsonwebtoken": "^9.0.2"
+    }
+  }
```
Comment (oleggromov) [inline package.json]: No newlines anywhere. This could be fixed either in your editor, or in `.gitconfig` (https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings).

File: index.js (hunk 1 above)
Comment (albertoperdomo) [reply index.js:22]: Yes, easy to understand is what we are aiming for!

Comment (albertoperdomo) [reply index.js:22]: I will replace var for const.

File: package.json (hunk 2 above)
Comment (albertoperdomo) [reply package.json]: Does fixing it in .gitconfig change it retroactively? I will look into changing my VS code settings for this.
I think it came from copy/paste from another GH repo.

Comment (oleggromov) [reply package.json]: I don't think so Alberto. The commit has been already made and pushed, so you'll have to save files with proper line endings and commit them again. Where these changes will originate from - gitconfig, your IDE or even manual editing - won't matter.

**Review by NevRA — APPROVED** [review]
LGTM

File: README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+1. Sign in to your Metabase instance as an admin. 
+2. Go to admin settings and enabling embedding.
+3. Under admin settings/embedding, click on static embedding and copy the embedding secret key.
+
+## Configure the app
+
+1. Paste the secret key into an env var: `$ export METABASE_EMBEDDING_SECRET="PASTE_SECRET_HERE"`
+2. Create an env var pointing to your Metabase site URL, if it's not on http://localhost:3000 `$ export METABASE_SITE_URL="http://localhost:4000"`
+3. Create an env var with the ID of the dashboard to embed: `$ export METABASE_EMBED_DASHBOARD_ID="8"`
+3. Install dependencies: `$ bundle install`
```
Comment (WiNloSt) [inline README.md:23]: I assume this is for Ruby? Also we already have that step in the next section on the 3rd one. 
> Install the dependencies: `$ npm install`

Maybe this step could be removed? What do you think?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+## Embed the dashboard
+1. Go to your dashboard, click on the share/embed button at the top
+2. Click on the "Publish" button
+3. Install the dependencies: `$ npm install`
+4. Start the server: `$ node index.js`
+5. Open the app in your browser at http://localhost:8080
```
Comment (WiNloSt) [inline README.md:30]: Just tested locally, interestingly if you run Metabase on dev mode, our webpack dev server will run on a default port 8080 making this example not working. Not sure how big of a deal that is though, but what I had to change the example app to run on other port instead of 8080 to test with my Metabase instance on dev mode.

Comment (albertoperdomo) [reply README.md:30]: This is mostly aimed at customers/users, so we can assume they are rather working with the downloaded JAR file or Docker image. I guess this is mostly an inconvenience internally, but I could still switch to any other port.

Do you have any suggestion for a particular number?

File: README.md (hunk 3 above)
Comment (albertoperdomo) [reply README.md:23]: Yes, this is an oversight. And yes there is a duplicate step to install the dependencies.
I will clean it up.

File: README.md (hunk 4 above)
Comment (WiNloSt) [reply README.md:30]: Another whole number port could be a good option since they look nicer e.g. 4000, or 5000 though 4000 is used by Cypress, but I won't expect folks to run both Cypress and embedding at the same time so that might not be big of a deal.

Other options people used is to run the digits from the based port e.g. our dev server is running at 3000 and we could use 3001, 3002, 3003, etc. I have the [embedding app port set to 3001](https://github.com/WiNloSt/metabase-embeddings-101/blob/main/package.json#L6) with no particular reason.

tl;dr either 3xxx, or 4000 is fine IMO.
